### PR TITLE
Remove `os.chdir()` calls

### PR DESCRIPTION
**Pull request by Anthchirp**
The current working directory is a process-wide singleton.
Messing with it can cause interesting behaviour.

Also allow the use of pathlib objects in a few more places.

### CODE REVIEW

File: format/Format.py [hunk 1]
```diff
@@ -317,6 +317,9 @@ def get_imageset(
         from dxtbx.imageset import ImageSequence, ImageSet, ImageSetData
 
         # Get filename absolute paths, for entries that are filenames
+        input_filenames = [
+            x.__fspath__() if hasattr(x, "__fspath__") else x for x in input_filenames
```
Comment (ndevenish) [inline format/Format.py:321]:
```suggestion
            os.fspath(x) for x in input_filenames
```

File: model/experiment_list.py [hunk 2]
```diff
@@ -1118,7 +1118,8 @@ def _openingpathiterator(pathnames: Iterable[str]):
     while paths:
         # Get the next path from the queue
         (do_recurse, pathname) = paths.popleft()
-
+        if pathname and hasattr(pathname, "__fspath__"):
+            pathname = pathname.__fspath__()
```
Comment (ndevenish) [inline model/experiment_list.py:1122]:
```suggestion
        if pathname:
            pathname = os.fspath(pathname)
```

File: sequence_filenames.py [hunk 3]
```diff
@@ -126,6 +126,8 @@ def group_files_by_imageset(filenames):
     # filename itself.
     template = []
     for f in filenames:
+        if f and hasattr(f, "__fspath__"):
+            f = f.__fspath__()
```
Comment (ndevenish) [inline sequence_filenames.py:130]:
```suggestion
        if f:
            f = os.fspath(f)
```
Although, I'm not sure the reason for the `if f:` everywhere - is it sometimes empty in these routines?

**Review by ndevenish — CHANGES_REQUESTED** [review]
We didn't use this pattern before because we needed to support python < 3.6

File: format/Format.py (hunk 1 above)
Comment (Anthchirp) [reply format/Format.py:321]: `x` can be a URL here, so am not sure this is appropriate here

Comment (ndevenish) [reply format/Format.py]: Ah, a URL object? Or just a string? because strings will be preserved....

**Pull request merged**